Automattic/jetpack
# Move react state site benefits request from PHP to React

The Jetpack dashboard is slow to load and has been traced partly to the PHP methods that populate some initial React state before serving the page. In particular, the request for `site benefits` has been identified as a particular culprit.

#### Changes proposed in this Pull Request:
This PR moves the request for site benefits out of PHP on the server, and the React side requests it asynchronously after loading.

Only the Disconnect Dialog currently uses the site benefits data.

#### Jetpack product discussion
p1HpG7-ere-p2#comment-52533

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
Regression test to make sure the disconnect dialog still works on both the Jetpack Dashboard and from the Plugins page
- checkout this branch and build plugins/jetpack, and access your dev environment i.e. *.jurassic.tube/
- go to the Jetpack Dashboard and click `Manage site connection` and ensure the disconnect modal still works
- go to the Plugins page and `Disconnect and deactivate` the Jetpack plugin and ensure the disconnect modal still works

Note that if following through and fully testing the disconnect step, the disconnect survey step only shows from the Jetpack dashboard, not from Plugins page. Only the first step has changed however.

If the connection is slow to return site benefits data the following fall back text will be displayed:
<img width="718" alt="fall_back" src="https://user-images.githubusercontent.com/8742105/154708831-08de4c10-be67-4716-9688-523c4f3ff2fa.png">
